vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_unblinded.cc

```diff
@@ -471,18 +596,77 @@ void Unblinded::Retry(
     return;
   }
 
-  if (contribution->step == ledger::ContributionStep::STEP_START) {
-    Start(types, contribution->contribution_id, callback);
-    return;
+  switch (contribution->step) {
+    case ledger::ContributionStep::STEP_START: {
+      Start(types, contribution->contribution_id, callback);
+      return;
+    }
+    case ledger::ContributionStep::STEP_PREPARE: {
+      ProcessTokens(types, contribution->contribution_id, callback);
+      return;
+    }
+    case ledger::ContributionStep::STEP_RESERVE: {
+      // Retrieve reserved unblinded tokens from cache for retry
+      // attempt
+      std::vector<ledger::UnblindedToken> list;
```

Comment:
we already have them reserved in db, so we should just query them from db

Reply:
Makes sense, fixed.